tests/framework/pkg/wait/watch.go

```diff
@@ -0,0 +1,33 @@
+package wait
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type WatchCheckFunc func(watch.Event) (bool, error)
+
+func WatchWait(watchInterface watch.Interface, check WatchCheckFunc) error {
```

Comment:
Can you add a docstring to explain what this does?

Reply:
Added.